boefjes/boefjes/plugins/kat_webpage_analysis/main.py

```diff
@@ -66,13 +67,36 @@ def run(boefje_meta: BoefjeMeta) -> list[tuple[set, bytes | str]]:
         if content_type[0] in ALLOWED_CONTENT_TYPES:
             body_mimetypes.add(content_type[0])
 
+    # in case of a full response object, we hexdump to avoid issues with binary data or different encoding
+    response_dump = json.dumps(create_response_object(response))
+    content = binascii.hexlify(response.content).decode()
```

Comment:
I missed this the first time, but doing a hexdump will increase the storage requirements. Bytes support binary data, so if we can avoid this it would be better.

Reply:
I've thought about this, but the problem I had was differentiating between the encoding formats for the response meta and response content (which could be different). I agree on the size though and it's probably worth it to spend some little time to find come up with a better solution.

Reply:
Base64?

Reply:
> Base64?

Base64 and other more efficient encodings could be interesting, but I'm currently trying something out that doesn't change the content